### PR TITLE
Add MLflow Tracking Quickstart

### PR DESCRIPTION
**Pull request by BenWilson2**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/BenWilson2/mlflow/pull/10285?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10285/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10285
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Adds an actual MLflow Tracking 5-minute quickstart and notebook 
Deprioritizes and explains the longer 6 page guide / tutorial

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [X] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [X] Yes. I've updated:
  - [X] Examples
  - [ ] API references
  - [X] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

Add an MLflow Tracking Quickstart

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [X] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by BenWilson2** [comment]
@chenmoneygithub here's the quickstart that I was talking about :) LMK what you think about the pacing, depth of content, and how it's presented with navigation elements!

**Comment by BenWilson2** [comment]
@mlflow-automation autoformat

File: docs/source/getting-started/index.rst [hunk 1]
```diff
@@ -6,11 +6,66 @@ quickstart tutorials here are the perfect starting point. They will guide you
 step-by-step through fundamental concepts, focusing purely on a task that will maximize your understanding of 
 how to use MLflow to solve a particular task. 
 
+`5-minute Quickstart - MLflow Tracking <intro-quickstart/index.html>`_
+----------------------------------------------------------------------
+
+In this brief introductory quickstart on MLflow Tracking, you will learn how to log, register, and load a model for inference. In the process of
```
Comment (chenmoneygithub) [inline docs/source/getting-started/index.rst:12]:
```suggestion
In this brief introductory quickstart on MLflow Tracking, you will learn how to log training stats (loss, accuracy, etc), then register your model on MLflow server, and finally load the registered model for inference. In the process of 
```

File: docs/source/getting-started/index.rst [hunk 2]
```diff
@@ -6,11 +6,66 @@ quickstart tutorials here are the perfect starting point. They will guide you
...
+
+.. raw:: html
+
+    <a href="https://raw.githubusercontent.com/mlflow/mlflow/master/docs/source/getting-started/intro-quickstart/notebooks/tracking_quickstart.ipynb" class="notebook-download-btn">
+        <i class="fas fa-download"></i>Download the Notebook</a><br>
+
+Quickstart elements
+^^^^^^^^^^^^^^^^^^^
+
+You can read through the quickstart directly as a guide, or navigate directly to the notebook example to get started.
```
Comment (chenmoneygithub) [inline docs/source/getting-started/index.rst:35]:
```suggestion
You can read through the quickstart as a guide, or navigate directly to the notebook example to get started. 
```

Minor nit: avoid double directly.

File: docs/source/getting-started/intro-quickstart/index.rst [hunk 3]
```diff
@@ -0,0 +1,229 @@
+MLflow Tracking Quickstart
+==========================
+
+Welcome to MLflow! 
+
+The purpose of this quickstart is to provide a quick, no-nonsense, bare-bones guide to the most essential core APIs of MLflow Tracking.
```
Comment (chenmoneygithub) [inline docs/source/getting-started/intro-quickstart/index.rst:6]:
```suggestion
The purpose of this quickstart is to provide a quick guide to the most essential core APIs of MLflow Tracking. 
```

Comment (chenmoneygithub) [reply docs/source/getting-started/intro-quickstart/index.rst:6]: minor nit: might keep it formal, I like the adjectives though.

File: docs/source/getting-started/intro-quickstart/index.rst [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+What you will learn
+-------------------
+
+In just a few minutes of following along with this quickstart, you will learn:
+
+* How to **log** parameters, metrics, and a model
+* The basics of the **MLflow fluent API**
+* How to **register** a model during logging 
+* How to navigate to a model in the **MLflow UI** 
+* How to **load** a logged model as a PyFunc for inference
```
Comment (chenmoneygithub) [inline docs/source/getting-started/intro-quickstart/index.rst:23]: I am not sure if people care about the concept "PyFunc" at model loading. Shall we instead say "How to load a logged model for inference"?

File: docs/source/getting-started/intro-quickstart/index.rst [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+            signature = infer_signature(X_train, lr.predict(X_train))
+
+            # Log the model
+            model_info = mlflow.sklearn.log_model(
+                sk_model=lr, 
+                artifact_path="iris_model", 
+                signature=signature, 
+                input_example=X_train, 
+                registered_model_name="tracking-quickstart"
+                )
```
Comment (chenmoneygithub) [inline docs/source/getting-started/intro-quickstart/index.rst:151]:
```suggestion
            )
```

File: docs/source/getting-started/intro-quickstart/index.rst [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+.. code-section::
+    .. code-block:: bash
+        :name: tracking-server-start
+
+        mlflow server --host 127.0.0.1 --port 8080
+
+.. note::
+    You can choose any port that you would like, provided that it's not already in use. 
+
+Step 3 - Log and Register a Model with MLflow
```
Comment (chenmoneygithub) [inline docs/source/getting-started/intro-quickstart/index.rst:64]: Shall we break down the code to smaller sections and briefly explain what each part is doing?

**Review by chenmoneygithub — CHANGES_REQUESTED** [review]
Thanks Ben, this is very well written! 

I only have one main suggestion - for the code block of training the model and logging, can we break it down to a few smaller sections with explanation? Imagining myself as a new user, I will be a bit confused about what I am doing with `mlflow.start_run()` and `infer_signature`.

File: docs/source/getting-started/intro-quickstart/index.rst (hunk 6 above)
Comment (BenWilson2) [reply docs/source/getting-started/intro-quickstart/index.rst]: Split it into 2 sections, added a note about why we're splitting the code there, and cleaned up some of the text. We want to avoid too much explanation in this quickstart, though. That's what the longer 6-page guide is for. This one is targeting the 'I just want to see this work in < 5 minutes" crowd.

File: docs/source/getting-started/intro-quickstart/index.rst (hunk 4 above)
Comment (BenWilson2) [reply docs/source/getting-started/intro-quickstart/index.rst]: removed

File: docs/source/getting-started/intro-quickstart/index.rst (hunk 3 above)
Comment (BenWilson2) [reply docs/source/getting-started/intro-quickstart/index.rst]: removed

File: docs/source/getting-started/index.rst (hunk 2 above)
Comment (BenWilson2) [reply docs/source/getting-started/index.rst]: good catch

File: docs/source/getting-started/index.rst (hunk 1 above)
Comment (BenWilson2) [reply docs/source/getting-started/index.rst]: Changed to a bullet list to simplify the long sentence

File: docs/source/getting-started/intro-quickstart/index.rst [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+
+.. code-section::
+    .. code-block:: python
+        :name: train-model
+
+        import mlflow
+        import mlflow.sklearn
+        from mlflow.models import infer_signature
+
+        import pandas as pd
```
Comment (harupy) [inline docs/source/getting-started/intro-quickstart/index.rst:82]:
```suggestion
```

File: docs/source/getting-started/intro-quickstart/index.rst [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+- Train a Logistic Regression model and evaluate its performance.
+- Prepare the model hyperparameters and calculate metrics for logging.
+
+
+.. code-section::
+    .. code-block:: python
+        :name: train-model
+
+        import mlflow
+        import mlflow.sklearn
```
Comment (harupy) [inline docs/source/getting-started/intro-quickstart/index.rst:79]:
```suggestion
```

File: docs/source/getting-started/intro-quickstart/index.rst [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+
+        # Split the data into training and test sets
+        X_train, X_test, y_train, y_test = train_test_split(
+            X, y, test_size=0.2, random_state=42
+        )
+
+        # Define the model hyperparameters
+        params = {
+            "solver": "lbfgs",
+            "max_iter": 1000,
```
Comment (harupy) [inline docs/source/getting-started/intro-quickstart/index.rst:103]:
```suggestion
            "max_iter": 10,
```

Comment (harupy) [reply docs/source/getting-started/intro-quickstart/index.rst:103]: can we remove solver and multi_class?

File: docs/source/getting-started/intro-quickstart/index.rst [hunk 10]
```diff
@@ -0,0 +1,248 @@
...
+        lr.fit(X_train, y_train)
+
+        # Predict on the test set
+        y_pred = lr.predict(X_test)
+
+        # Calculate metrics
+        accuracy = accuracy_score(y_test, y_pred)
+        precision = precision_score(y_test, y_pred, average="weighted")
+        recall = recall_score(y_test, y_pred, average="weighted")
+        f1 = f1_score(y_test, y_pred, average="weighted")
```
Comment (harupy) [inline docs/source/getting-started/intro-quickstart/index.rst:119]: Can we only log `accuracy`?

File: docs/source/getting-started/intro-quickstart/index.rst [hunk 11]
```diff
@@ -0,0 +1,248 @@
...
+        from mlflow.models import infer_signature
+
+        import pandas as pd
+        from sklearn import datasets
+        from sklearn.model_selection import train_test_split
+        from sklearn.linear_model import LogisticRegression
+        from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score
+
+        # Set our tracking server uri for logging
+        mlflow.set_tracking_uri(uri="http://127.0.0.1:8080")
```
Comment (harupy) [inline docs/source/getting-started/intro-quickstart/index.rst:89]: This line can be moved to the next section.

File: docs/source/getting-started/intro-quickstart/index.rst [hunk 12]
```diff
@@ -0,0 +1,248 @@
...
+        y_pred = lr.predict(X_test)
+
+        # Calculate metrics
+        accuracy = accuracy_score(y_test, y_pred)
+        precision = precision_score(y_test, y_pred, average="weighted")
+        recall = recall_score(y_test, y_pred, average="weighted")
+        f1 = f1_score(y_test, y_pred, average="weighted")
+
+        # Create a new MLflow Experiment
+        mlflow.set_experiment("MLflow Quickstart")
```
Comment (harupy) [inline docs/source/getting-started/intro-quickstart/index.rst:122]: This line can be moved to the next section.

File: docs/source/getting-started/intro-quickstart/index.rst
Comment (harupy) [inline docs/source/getting-started/intro-quickstart/index.rst:1]: Is it possible to create `docs/source/getting-started/intro-quickstart/notebooks/tracking_quickstart.ipynb` from this file?

File: docs/source/getting-started/intro-quickstart/index.rst [hunk 13]
```diff
@@ -0,0 +1,248 @@
...
+--------------------------------
+
+We're going to start a local MLflow Tracking Server, which we will connect to for logging our data for this quickstart.
+From a terminal, run:
+
+.. code-section::
+    .. code-block:: bash
+        :name: tracking-server-start
+
+        mlflow server --host 127.0.0.1 --port 8080
```
Comment (harupy) [inline docs/source/getting-started/intro-quickstart/index.rst:59]:
```suggestion
.. code-block:: bash
    :name: tracking-server-start

    mlflow server --host 127.0.0.1 --port 8080
```

Can we remove `code-section:`?

File: docs/source/getting-started/intro-quickstart/index.rst [hunk 14]
```diff
@@ -0,0 +1,248 @@
...
+        from sklearn.model_selection import train_test_split
+        from sklearn.linear_model import LogisticRegression
+        from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score
+
+        # Set our tracking server uri for logging
+        mlflow.set_tracking_uri(uri="http://127.0.0.1:8080")
+
+        # Load the Iris dataset
+        iris = datasets.load_iris()
+        X, y = iris.data, iris.target
```
Comment (harupy) [inline docs/source/getting-started/intro-quickstart/index.rst:93]:
```suggestion
        X, y = datasets.load_iris(return_X_y=True)
```

File: docs/source/getting-started/intro-quickstart/index.rst (hunk 7 above)
Comment (BenWilson2) [reply docs/source/getting-started/intro-quickstart/index.rst:82]: We use this in step 5 in crafting the `result` DataFrame

File: docs/source/getting-started/intro-quickstart/index.rst (hunk 10 above)
Comment (BenWilson2) [reply docs/source/getting-started/intro-quickstart/index.rst:119]: sure. I'll update.

File: docs/source/getting-started/intro-quickstart/index.rst
Comment (BenWilson2) [reply docs/source/getting-started/intro-quickstart/index.rst:1]: The explanation content is not in a format that is compatible uniformly between rst and jupyter, as well as additional notes that won't render properly or look particularly nice / attention-grabbing in a notebook.

File: docs/source/getting-started/intro-quickstart/index.rst (hunk 13 above)
Comment (BenWilson2) [reply docs/source/getting-started/intro-quickstart/index.rst:59]: All code blocks throughout the docs are going to be converted to use language-specific indicators to match the style of Databricks docs. The code-section formatting is a consistent formatting going forward which will make much more sense as examples for R are added (so that code blocks are selectable to change the displayed code within the docs).

File: docs/source/getting-started/intro-quickstart/index.rst (hunk 9 above)
Comment (harupy) [reply docs/source/getting-started/intro-quickstart/index.rst:103]: Got it. If it's fast, that's fine.

File: docs/source/getting-started/intro-quickstart/index.rst
Comment (harupy) [reply docs/source/getting-started/intro-quickstart/index.rst:1]: Got it. Any idea to prevent mismatches between rst and ipynb?

File: docs/source/getting-started/intro-quickstart/index.rst (hunk 13 above)
Comment (harupy) [reply docs/source/getting-started/intro-quickstart/index.rst:59]: sounds good :)

File: docs/source/getting-started/intro-quickstart/index.rst
Comment (BenWilson2) [reply docs/source/getting-started/intro-quickstart/index.rst:1]: Convert our docs to markdown? Although, that's a bit besides the point. We're not going to convert all documentation tutorials into pure notebook format for the sake of expediency. The instructional formats simply don't always (which isn't to say that they won't, since the flavor-specific tutorials that I'm working on right now are going to be notebook-only with no 'teaching' page associated with them) conform to the same demonstrative style that a notebook has. 
The notebooks should be very lean, without an excessive amount of text in markdown cells in order to make them readable, quick-to-execute locally, and efficient to execute and see the results within the UI. Some concepts for teaching (particularly the early stage introductory tutorials) can benefit from longer explanations with notes / warnings / callouts and diagrams that simply don't work that well with a notebook format.

**Review by chenmoneygithub — APPROVED** [review]
